crates/bevy_pbr/src/light_probe/environment_map.rs

```diff
@@ -446,40 +330,13 @@ fn add_texture_view<'a>(
     }
 }
 
-#[cfg(all(
-    not(feature = "shader_format_glsl"),
-    not(target_arch = "wasm32"),
-    not(target_os = "ios"),
-    not(target_os = "android")
-))]
-impl<'a> RenderViewBindGroupEntries<'a> {
-    /// Returns a list of texture views of each diffuse cubemap, in binding
-    /// order.
-    pub(crate) fn diffuse_texture_views(&'a self) -> &'a [&'a <TextureView as Deref>::Target] {
-        self.diffuse_texture_views.as_slice()
-    }
-
-    /// Returns a list of texture views of each specular cubemap, in binding
-    /// order.
-    pub(crate) fn specular_texture_views(&'a self) -> &'a [&'a <TextureView as Deref>::Target] {
-        self.specular_texture_views.as_slice()
-    }
-}
-
-#[cfg(any(
-    feature = "shader_format_glsl",
-    target_arch = "wasm32",
-    target_os = "ios",
-    target_os = "android"
-))]
-impl<'a> RenderViewBindGroupEntries<'a> {
-    /// Returns the texture view corresponding to the view's diffuse cubemap.
-    pub(crate) fn diffuse_texture_views(&self) -> &'a TextureView {
-        self.diffuse_texture_view
-    }
+impl RenderDeviceExt for RenderDevice {
+    fn binding_arrays_are_available(&self) -> bool {
+        use bevy_render::settings::WgpuFeatures;
 
-    /// Returns the texture view corresponding to the view's specular cubemap.
-    pub(crate) fn specular_texture_views(&self) -> &'a TextureView {
-        self.specular_texture_view
+        !cfg!(feature = "shader_format_glsl")
```

Comment:
I don't think testing on this feature is right. You can enable it because you have one shader in glsl, but still render on vulkan or whatever and have all your other shaders in wigs

Reply:
This was discussed before. Naga_oil compiles every shader module in GLSL to test that names are valid (at least in debug mode). So even if a shader is never used in GLSL it will still try to compile `environment_map.wgsl` in GLSL and end up crashing because GLSL doesn't support binding arrays. @robtfm's preferred approach was to just `cfg` it off using Cargo features.

Reply:
could you add a comment explaining that?

Reply:
Done.